src/foam/nanos/auth/resetPassword/ResetPasswordByCode.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth.resetPassword',
+  name: 'ResetPasswordByCode',
+  extends: 'foam.nanos.auth.resetPassword.ResetPassword',
+
+  documentation: 'Reset Password By Code Model',
+
+  imports: [
+    'ctrl',
+    'resetPasswordService'
+  ],
+
+  requires: [
+    'foam.log.LogLevel',
+    'foam.u2.dialog.NotificationMessage',
+    'foam.u2.FragmentedTextField',
+    'foam.u2.FragmentedTextFieldFragment'
+  ],
+
+  messages: [
+    { name: 'CODE_SECTION_TITLE', message: 'Please enter your email verification code' },
+    { name: 'RESET_PASSWORD_TITLE', message: 'Reset your password' },
+    { name: 'RESET_PASSWORD_SUBTITLE', message: 'Create a new password for your account' }
+  ],
+
+  sections: [
+    {
+      name: 'verificationCodeSection',
+      title: function() {
+        return this.CODE_SECTION_TITLE;
```

Comment:
Is this needed? Can't we set section titles directly from the locale system, just like property labels?